lib/datadog/core/remote/client.rb

```diff
@@ -117,7 +117,17 @@ def payload
             tracer_version: Core::Environment::Identity.tracer_version,
             service: Datadog.configuration.service,
             env: Datadog.configuration.env,
-            tags: [], # TODO: add nice tags!
+            tags: [
+              format('ruby.tracer.version:%s', Core::Environment::Identity.tracer_version),
+              format('ruby.runtime.platform:%s', RUBY_PLATFORM),
+              format('ruby.runtime.version:%s', RUBY_VERSION),
+              format('ruby.runtime.engine.name:%s', RUBY_ENGINE),
+              format(
+                'ruby.runtime.engine.version:%s',
+                (defined?(RUBY_ENGINE_VERSION) ? RUBY_ENGINE_VERSION : RUBY_VERSION)
+              ),
+              format('ruby.rubygems.platform.local:%s', Gem::Platform.local.to_s),
```

Comment:
`ruby.rubygems.platform.local` or `ruby.gem.platform.local`? I'm not sure.

Reply:
Technically the `Gem::Platform` class is part of the rubygems. But it is also true that `ruby.gem.platform.local` seems more comfortable to read. 

Naming is hard. If we use the tags to help filter on the dashboards, I would use `ruby.gem.platform.local` to simplify typing less characters.

Would it be possible to have a case where `RUBY_PLATFORM` and `Gem::Platform.local.to_s` differ? Asking, because if that is not the case, we might be able to remove one of those 😄

Reply:
Haha I had the same thoughts, but ultimately I was thinking of using the `ruby.gem.` prefix for _actual gems_, so the component after would be the gem name: `ruby.gem.<gem name>.<gem property>`. 

Examples:

- `ruby.gem.libddwaf.version`
- `ruby.gem.libddwaf.platform` # could be "from source" `ruby` gem platform or a binary gem platform like `arm64-darwin`
- `ruby.gem.libddwaf.loaded` # or something like that. Use case is: currently this should be `false` when above is `ruby` gem platform since we bail out and don't build from source, therefore even though the gem is there it's failing to provide any facility, so ASM is inoperative.
- `ruby.gem.libdatadog.version`
- `ruby.gem.libdatadog.platform`

So with this distinction you may understand why I put `ruby.rubygems.` there, `Gem::Platform` being actually part of `rubygems`.

Reply:
> Would it be possible to have a case where RUBY_PLATFORM and Gem::Platform.local.to_s

Yes, it is important to have both to be able to distinguish these cases (and many others):

- On old Ruby, `RUBY_PLATFORM == 'x86_64-linux'` even on `musl`
- On old rubygems, `RUBY_PLATFORM == 'x86_64-linux-musl'` becomes `Gem::Platform.local.to_s == 'x86_64-linux'`
- On darwin, `RUBY_PLATFORM == 'x86_64-darwin22'` becomes `Gem::Platform.local.to_s == 'x86_64-darwin-22'`
- On Alpine's apk ruby package, `RUBY_PLATFORM == 'x86_64-linux-musl'` but `Gem::Platform.local == 'ruby'`, forcing every gem to install from source